### PR TITLE
Initial implementation of translation strings compression

### PR DESCRIPTION
**Pull request by alvinhochun**
* **Please check if the PR fulfills these requirements**
- ✔️  The changes have been tested locally
- ✔️  There are no breaking changes

* **What kind of change does this PR introduce?**

Implement translation strings compression (not built by default)

* **Other information**:

Following #926, my working tree was getting a bit messy and I don't want to pile up too many changes at once, so I am making this PR.

Things to note:

- The changes should not have affected the normal uncompressed builds.
- The compressed translations are generated from the normal translations.
    - It first compiles the translation cpp file without LTO, then dumps the raw structure data from the object file using `objcopy`; this is to make sure that the data is in the correct data layout. The data is then compressed and a new translation cpp file is generated from it.
    - The `make_translation.py` serializes the language translation data to be reused when generating the compressed translation cpp file, to make sure that the font table symbols order is consistent.
- The firmwares with compressed string data can be built using the `firmware-string_compressed_${LANG}` target.
- The compressed firmwares are not built by CI.
- I am using https://github.com/janding/lzfx compression as a proof-of-concept, but I might change to a different algorithm.

### CODE REVIEW

File: Translations/make_translation.py [hunk 1]
```diff
@@ -397,29 +406,134 @@ def escape(string: str) -> str:
...
+
+    if not lzfx_strings:
+        translation_strings_and_indices_text = get_translation_strings_and_indices_text(
+            lang, defs, symbol_conversion_table
+        )
+        f.write(translation_strings_and_indices_text)
+        f.write(
+            "const TranslationIndexTable *const Tr = &TranslationIndices;\n"
+            "const char *const TranslationStrings = TranslationStringsData;\n\n"
+            "extern const uint8_t *const Font_12x16 = USER_FONT_12;\n"
```
Comment (Ralim) [inline Translations/make_translation.py:484]: Could pull out the duplicate two lines here outside of the if

**Comment by Ralim** [comment]
I'm curious why you went down the path of parsing the produced translation file and then compressing rather than just doing the compression without the extra compile stage; is this just so that the compiler strips unused strings for you?

( I may be missing something; haven't had coffee yet)

**Comment by alvinhochun** [comment]
I just think it's safer. I don't need to worry about getting the endianness and field ordering wrong building the binary data myself. By taking the data straight from the object file, I can be certain that the decompressed bytes matches what the code expects. If the struct field ordering does not match the defined order in the header, the compiler will yell at me about it.

The compiler can't strip anything there, unfortunately.

**Comment by Ralim** [comment]
Ah that does make total sense.

Not the way I would have coded it, but I like it :)

Can you sync to master and I'm happy to merge :)

**Review by Ralim — APPROVED** [review]

**Pull request merged**